### PR TITLE
SAIC-543: Cinder volume migration

### PR DESCRIPTION
**Pull request by aguzikova**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_filter
cloudferrylib/os/actions/check_filter.py:39: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:43: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:46: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:54: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:58: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:61: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:70: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:75: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:78: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:87: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:92: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/actions/check_filter.py:95: [W1202(logging-format-interpolation), CheckFilter.run] Use % formatting in logging functions but pass the % parameters as arguments
************* Module cloudferrylib.os.storage.cinder_database
cloudferrylib/os/storage/cinder_database.py:223: [W1202(logging-format-interpolation), CinderStorage.deploy_data_to_table.filter_data] Use % formatting in logging functions but pass the % parameters as arguments
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:95: [W0221(arguments-differ), CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:153: [W1201(logging-not-lazy), CinderStorage.finish] Specify string format arguments as logging function parameters
cloudferrylib/os/storage/cinder_storage.py:252: [W0613(unused-argument), CinderStorage.convert_snapshot] Unused argument 'cloud'
************* Module cloudferrylib.utils.remote_runner
cloudferrylib/utils/remote_runner.py:31: [W0621(redefined-outer-name), RemoteRunner.__init__] Redefining name 'sudo' from outer scope (line 15)
************* Module tests.generate_load
devlab/tests/generate_load.py:7: [F0401(import-error), ] Unable to import 'config'
devlab/tests/generate_load.py:9: [F0401(import-error), ] Unable to import 'filtering_utils'
devlab/tests/generate_load.py:36: [W0621(redefined-outer-name), clean_if_exists.wrapper] Redefining name 'args' from outer scope (line 1188)
devlab/tests/generate_load.py:971: [W0120(useless-else-on-loop), CleanEnv.clean_vms.wait_until_vms_all_deleted] Else clause on loop without a break statement
devlab/tests/generate_load.py:977: [W0106(expression-not-assigned), CleanEnv.clean_vms] Expression "[vms.append(vm) for vm in self.config.vms_from_volumes]" is assigned to nothing
devlab/tests/generate_load.py:1122: [W0141(bad-builtin), CleanEnv.clean_keypairs.delete_user_keypairs] Used builtin function 'map'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/attach_used_volumes_via_compute.py [hunk 1]
```diff
@@ -28,8 +30,11 @@ def run(self, info, **kwargs):
             if not instance[utl.META_INFO].get(utl.VOLUME_BODY):
                 continue
             for vol in instance[utl.META_INFO][utl.VOLUME_BODY]:
-                if storage_res.get_status(
-                        vol['volume']['id']) != 'in-use':
+                try:
+                    status = storage_res.get_status(vol['volume']['id'])
+                except NotFound:
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/attach_used_volumes_via_compute.py]: This doesn't look right. Each instance volumes must be migrated, if volume is not found in destination cloud - then we somehow missed it in the earlier stages. Please look through.

File: cloudferrylib/os/actions/check_filter.py [hunk 2]
```diff
@@ -46,6 +48,21 @@ def run(self, **kwargs):
...
+        if search_opts_vol and search_opts_vol.get('volumes_list'):
+            volumes_list = search_opts_vol['volumes_list']
+            for vol_id in volumes_list:
+                LOG.debug('Filtered image id: {}'.format(vol_id))
+                try:
+                    vol = cinder_resource.cinder_client.volumes.get(vol_id)
+                    if vol:
+                        LOG.debug('Filter config check: Volume ID {} is OK'
+                                  .format(vol_id))
+                except glance_exc.HTTPNotFound as e:
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/check_filter.py]: Catching glance exception on cinder client command will probably never work.

File: cloudferrylib/os/actions/cinder_database_manipulation.py [hunk 3]
```diff
@@ -47,18 +113,310 @@ def run(self, *args, **kwargs):
...
+        :return: list of paths
+
+        """
+        if vt:
+            cmd = (
+                "for exp in "
+                r"$(awk -F= '/\["
+                "%s"
+                r"\]/{i=1} /!/ {i=0} i && /nfs_shares_config/ "
+                "{print $2; exit}' /etc/cinder/cinder.conf | xargs cat); "
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: Cinder config may  be anywhere in the filesystem. Please make it configurable.

File: cloudferrylib/os/actions/cinder_database_manipulation.py [hunk 4]
```diff
@@ -47,18 +113,310 @@ def run(self, *args, **kwargs):
...
+
+            dstpaths = self._paths(DST, v['volume_type_id'])
+
+            src = self.find_dir(self.cloud[SRC], srcpaths, v)
+            dst = self.find_dir(self.cloud[DST], dstpaths, v)
+
+            if not dst:
+                dst = dstpaths
+            else:
+                dst = '/'.join(dst.split('/')[0:-1])
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/cinder_database_manipulation.py]: `os.path.dirname()`?

File: cloudferrylib/os/storage/cinder_storage.py [hunk 5]
```diff
@@ -19,8 +19,12 @@
 from cloudferrylib.base import storage
 from cloudferrylib.utils import utils as utl
 
+import re
+
 LOG = utl.get_log(__name__)
 
+RE_EXTRACT_HOST = re.compile('//([^:^/]*)')
```
Comment (roman-verchikov) [inline cloudferrylib/os/storage/cinder_storage.py]: Please use raw python strings for regex (`r'.*my_regex.*'`)

**Pull request closed without merging**